### ISSUE TITLE
Excalibur does not behave as expected during resizes when in fullscreen mode

### ISSUE THREAD

**Issue by eonarheim** [excaliburjs/Excalibur#1069]
Excalibur does not seem to be handling fullscreen resizes very well, cam

### Steps to Reproduce
See pen https://codepen.io/eonarheim/pen/oQmeLZ?editors=0010

Resize the window up and down and observe that the actors disappear and the platform does not stay at the bottom of the screen as expected. Actors are being marked as offscreen, the camera focus appears to change, and some odd scaling factor is being applied.

![image](https://user-images.githubusercontent.com/612071/49330916-25372480-f55b-11e8-8ba4-7b4d8eb1631b.png)

![image](https://user-images.githubusercontent.com/612071/49330921-31bb7d00-f55b-11e8-847b-6bbf4b2c11cc.png)

### Expected Result
* [ ] The platform in the above example should stay locked to the bottom of the screen.
* [ ] Drawing canvas should scale with resize in fullscreen mode 
![image](https://user-images.githubusercontent.com/612071/49330936-76471880-f55b-11e8-8995-0be3db1fc73b.png)
* [ ] Actors should not be marked offscreen during scaling like this

### Actual Result
All of the above

### Environment
- Excalibur versions: master

### Current Workaround
None

**Comment by justjoeyuk** [comment]
This is a fairly big issue. It’s a non-issue for me because I’m making a messenger game. However, resizing a browser window is very common.

Is it only in Fullscreen mode that this happens?

**Comment by eonarheim** [comment]
@justjoeyuk I've only noticed it in fullscreen, but any viewport modification I suspect may cause similar issues. Correct this may be a difficult issue to solve, I took a stab at it earlier and got maybe 1/2 way https://github.com/excaliburjs/Excalibur/commit/d38ddd1a8b59de31376b5f86f60ad15efd8cd978 Definitely feel free to take or leave anything I've done if you work on fixing this.

The steps to fix this may be:
 * Observe the viewport bounds for changes (maybe every frame or with a throttled window event)? 
 * Correctly re-configure engine parameters to adjust to the new bounds
   - Potentially refactor the `Engine.ts` display mode processing (currently its a mess in there ☹️) Ideally we could extract out the display mode behavior into separate files (maybe using [type object pattern?](http://gameprogrammingpatterns.com/type-object.html)) 
 * Ensure actors behave correctly on/off screen
